api/models/ml_model.py

```diff
@@ -59,49 +57,51 @@ def validate_image_size(cls, image_size):
-                                 f'"posenet". ')
+                                 f' {MODELS_DEVICES["EdgeTPU"]}. ')
+
         elif values.get("device") == "Dummy":
             # No restrictions on this model.
-            pass
-        elif values.get("device") in ["x86", "x86-gpu"]:
-            if values.get("name") not in ["mobilenet_ssd_v2", "openvino", "openpifpaf", "openpifpaf_tensorrt",
-                                          "yolov3"]:
+            # All available models.
+            if values.get("name") not in MODELS_DEVICES["Dummy"]:
+                raise ValueError(f'The device {values.get("device")} only supports the following models:'
+                                 f' {MODELS_DEVICES["Dummy"]}. ')
+
+        elif values.get("device") == "x86":
+            if values.get("name") not in MODELS_DEVICES["x86"]:
+                raise ValueError(f'The device {values.get("device")} only supports the following models:'
+                                 f' {MODELS_DEVICES["x86"]}. ')
+
+        elif values.get("device") == "x86-gpu":
+            if values.get("name") not in MODELS_DEVICES["x86-gpu"]:
                 raise ValueError(f'The device {values.get("device")} only supports the following models:'
-                                 f'"mobilenet_ssd_v2", "openvino","openpifpaf","openpifpaf_tensorrt"'
-                                 f'"yolov3". ')
+                                 f' {MODELS_DEVICES["x86-gpu"]}. ')
 
         return values
 
     @root_validator(skip_on_failure=True)
     def check_variables_for_models(cls, values):
 
-        if values.get("name") == "ssd_mobilenet_v2_coco" or values.get("name") == "ssd_mobilenet_v2_pedestrian_softbio":
+        if values.get("name") in ["ssd_mobilenet_v2_coco", "ssd_mobilenet_v2_pedestrian_softbio", "mobilenet_ssd_v2",
+                                  "pedestrian_ssd_mobilenet_v2", "pedestrian_ssdlite_mobilenet_v2", "mobilenet_ssd_v2",
+                                  "openvino"]:
             pass
 
         elif values.get("name") == "openpifpaf_tensorrt":
-            # image size x,x,y puede ser?
             if values.get("tensorrt_precision") is None:
                 raise ValueError('The model "openpifpaf_tensorrt" requires the parameter "tensorrt_precision", and'
                                  'said parameter can be either 16 or 32.')
 
-        elif values.get("name") == "mobilenet_ssd_v2":
-            pass
-
-        elif values.get("name") == "pedestrian_ssd_mobilenet_v2":
-            pass
-
-        elif values.get("name") == "pedestrian_ssdlite_mobilenet_v2":
-            pass
+            integers = values.get("image_size").split(",")
+            if int(integers[0]) % 16 != 1 or int(integers[0]) % 16 != 1:
+                raise ValueError('First two values of ImageSize must be multiples of 16 plus 1 for openpifpaf_tensorrt.'
```

Comment:
I think there are only 3 valid options but I don't think it is too important to have the exact validation here

Reply:
Do you suggest removing all validations related to image size?